src/librustc_errors/json/tests.rs

```diff
@@ -40,6 +39,13 @@ impl<T: Write> Write for Shared<T> {
     }
 }
 
+fn with_default_globals(f: impl FnOnce()) {
```

Comment:
Hmm; maybe this should be shared with libsyntax by moving it to libsyntax_pos?

Reply:
Hm, my impression at a quick glance was that there are two "globals" in separate TLS variables (one declared in syntax, and one in syntax_pos). I don't think we buy anything from doing that recombination... in any case, it seems like ideally we'd cleanup the globals but that should be done separately I think.

Reply:
Fair :)